src/content/docs/explanations/secrets/index.mdx

```diff
@@ -7,33 +7,52 @@ such as authentication tokens, passwords or even URLs.
 
 ## Usage in TQL
 
-There are two ways to pass an argument to a operator that expects a secret:
+Secret values can only be used by operators that accept secrets. Operators
+generally do not document that they accept a secret, but will accept secrets
+where appropriate.
 
-- By providing a plain `string` ([Ad-hod Secret](#ad-hoc-secrets)):
+There are two ways to pass an argument to a operator that expects a secret,
+shown here using the [`to_splunk`](/reference/operators/to_splunk) operator
+as an example. The operator expects a HEC-token for authentication. This gives
+you two options:
+
+- Providing a plain `string` ([Ad-hod Secret](#ad-hoc-secrets)):
 
   ```tql
   to_splunk "https://localhost:8088", hec_token="my-plaintext-token"
   ```
 
-- By using the [`secret`](/reference/functions/secret) function
+  Here, a "secret" containing the string literal `my-plaintext-token` is created.
+
+- Using the [`secret`](/reference/functions/secret) function
   ([Manged Secret](#managed-secrets)):
 
   ```tql
   to_splunk "https://localhost:8088", hec_token=secret("splunk-hec-token")
   ```
 
-  This will fetch the value of the secret named `splunk-hec-token`.
-
-Operators generally do not document that they accept a secret, but will accept
-secrets where appropriate.
+  The operator will then fetch the secret named `splunk-hec-token` to
+  authenticate with the Splunk endpoint.
 
 ## The `secret` type
 
 Secrets are a special type in Tenzir's [type system](/reference/language/types).
+They can only be created using the [`secret`](/reference/functions/secret)
+function.
+
+### Internals
+
+A value of type `secret` does not actually contain the true secret value, but
+only instructions on how to obtain the true value, for example the key to look
+up in the secret store. For Ad-Hoc secrets that were created from a string literal,
+the literal value is held.
```

Comment:
Even though it's called `Internals`, this feels a bit too developer-centric.

Maybe something like this? "A value of type secret does not contain the secret value, only the secret name. When an operator is using a secret, the is name resolved aynchronously by querying the Tenzir Platform. For ad-hoc secrets that were created from a string literal, the secret name and value are identical so no lookup takes place."